flake.nix

```diff
@@ -27,206 +28,267 @@ rec {
   };
 
   outputs =
-    { self
-    , nixpkgs
-    , flake-utils
-    , released-nixpkgs-unstable
-    , released-nixpkgs-stable
-    , released-nix-unstable
-    , released-nix-stable
-    , nix-pills
+    inputs@{
+      self,
+      flake-parts,
+      git-hooks-nix,
+      nixpkgs,
+      released-nixpkgs-unstable,
+      released-nixpkgs-stable,
+      released-nix-unstable,
+      released-nix-stable,
+      nix-pills,
+      systems,
     }:
-    flake-utils.lib.eachDefaultSystem (system:
-      let
-        overlay = final: prev: {
-          asciinema-scenario = final.rustPlatform.buildRustPackage rec {
-            pname = "asciinema-scenario";
-            version = "0.2.0";
-            src = final.fetchCrate {
-              inherit pname version;
-              sha256 = "sha256-qMGi+myppWBapM7TkPeXC2g/M1FA1YGwESNrx8LVXkw=";
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+        inputs.git-hooks-nix.flakeModule
+      ];
+      systems = import systems;
+      perSystem =
+        { config, system, ... }:
+        let
+          overlay = final: prev: {
+            asciinema-scenario = final.rustPlatform.buildRustPackage rec {
+              pname = "asciinema-scenario";
+              version = "0.2.0";
+              src = final.fetchCrate {
+                inherit pname version;
+                sha256 = "sha256-qMGi+myppWBapM7TkPeXC2g/M1FA1YGwESNrx8LVXkw=";
+              };
+              cargoSha256 = "0z4iwjm38xfgipl1pcrkl8277p627pls565k7cclrhxfcx3f513p";
             };
-            cargoSha256 = "0z4iwjm38xfgipl1pcrkl8277p627pls565k7cclrhxfcx3f513p";
           };
-        };
 
-        pkgs = import nixpkgs { inherit system; overlays = [ overlay ]; };
-        inherit (pkgs.lib) getVersion;
+          pkgs = import nixpkgs {
```

Comment:
pkgs is already provided by the flake.parts framework, use it as such: https://flake.parts/overlays#consuming-an-overlay